### PR TITLE
Adding GKE private cluster requirement

### PR DESCRIPTION
**Pull request by xeviknal**
related to https://github.com/kiali/kiali/issues/3494
Pretty much based on https://istio.io/latest/docs/setup/platform-setup/gke/

### CODE REVIEW

File: content/documentation/staging/faq/general/google-cloud-private-cluster.adoc [hunk 1]
```diff
@@ -0,0 +1,15 @@
+---
+title: What do I need to run Kiali in a private cluster?
+weight: 60
+---
+:icons: font
+:sectlinks:
+:linkattrs:
+
+Private clusters have higher network restrictions. Kiali needs your cluster to allow `TCP` traffic between the Kubernates API service and the Istio Control Plane namespace in both `8080` and `15000` ports. Features like `Health` and `Envoy Dump` won't be workinig as expected.
```
Comment (jshaughn) [inline content/documentation/staging/faq/general/google-cloud-private-cluster.adoc:9]: Kubernates --> Kubernetes
namespace in both --> namespace, for both the
Features like `Health` and `Envoy Dump` won't be workinig as expected -->  This is required for features such as  `Health` and `Envoy Dump` to work as expected.

File: content/documentation/staging/faq/general/google-cloud-private-cluster.adoc [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+Make sure that the firewalls in you cluster allow the connections mentioned before.
```
Comment (jshaughn) [inline content/documentation/staging/faq/general/google-cloud-private-cluster.adoc:11]: you --> your
before --> above

File: content/documentation/staging/faq/general/google-cloud-private-cluster.adoc [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+Check our section of Google Kubernetes Engine (GKE) Private Clusters requirements in the https://kiali.io/documentation/latest/installation-guide/#_google_cloud_private_cluster_requirements[Installation Guide, window="_blank"].
```
Comment (jshaughn) [inline content/documentation/staging/faq/general/google-cloud-private-cluster.adoc:13]: our section of --> section
Clusters -> Cluster

File: content/documentation/staging/faq/general/google-cloud-private-cluster.adoc [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+icon:bullhorn[size=1x]{nbsp} Open an https://github.com/kiali/kiali/issues/new/choose[issue, window="_blank"] if you have a private cluster with a different provider than GKE. We'll try to accomodate your scenario and document it for future users.
```
Comment (jshaughn) [inline content/documentation/staging/faq/general/google-cloud-private-cluster.adoc:15]: accomodate --> accommodate

File: content/documentation/staging/installation-guide/index.adoc [hunk 5]
```diff
@@ -84,6 +84,24 @@ If you are installing on OpenShift, you must grant the `cluster-admin` role to t
 
 icon:lightbulb[size=1x]{nbsp}For most commands listed on this page the Kubernetes CLI command `kubectl` is used to interact with the cluster environment. On OpenShift you can simply replace `kubectl` with `oc`, unless otherwise noted.
 
+=== Google Cloud Private Cluster Requirements
+
+Private clusters on Google Cloud have network restrictions. Kiali needs that your cluster give access from the Kubernetes API to the Istio Control Plane namespace in both `8080` and `15000` ports. In order to do that, update your firewall with those two ports.
```
Comment (jshaughn) [inline content/documentation/staging/installation-guide/index.adoc:89]: needs that your cluster give access -->  needs your cluster to allow access
namespace in --> namespace, for both the
 In order to do that, update -->  You must update

**Review by jshaughn — CHANGES_REQUESTED** [review]
Thanks @xeviknal, good content.  Just a few changes for staging, that can then be copied down to the older versions.

**Comment by xeviknal** [comment]
@jshaughn comments approached!
I feel like a school kid who doesn't read his homework before handing it in :sweat:  Thanks for the corrections.
I've changed all the versions that the info applies to.

**Review by jshaughn — APPROVED** [review]
> I feel like a school kid who doesn't read his homework before handing it in

Are you kidding!  You get an 'A' for being able to write so well in a second language.

**Comment by dex4er** [comment]
Hi. It is worth of mentioning that Istio by itself requires additional ports to be opened: 15017 (Injection and validation service for GCP loadbancer healthcheck) and 15014 (Control plane monitoring for `istioctl x` commands).

**Comment by xeviknal** [comment]
hi @dex4er,

thanks for commenting. I am checking the [official istio installation page for GKE](https://istio.io/latest/docs/setup/platform-setup/gke/) and they don't mention the same ports as you do. Are those ports new for istio 1.9? I can't see them anywhere.

Anyway, I am going to add the mention to the istio installation for GKE page to be more helpful, as you suggested.

thanks

File: content/documentation/staging/installation-guide/index.adoc [hunk 6]
```diff
@@ -102,6 +103,7 @@ To replace the existing rule and allow master access:
 gcloud compute firewall-rules update <firewall-rule-name> --allow <previous-ports>,tcp:8080,tcp:15000
 ----
 
+icon:bullhorn[size=1x]{nbsp} Istio deployments on private clusters also need extra ports to be opened. Check the https://istio.io/latest/docs/setup/platform-setup/gke/[Istio installation page for GKE,window="_blank"] to see all the extra installation steps for this platform.
```
Comment (xeviknal) [inline content/documentation/staging/installation-guide/index.adoc:106]: Good Monday @jshaughn ! Your English student has another delivery for you. Do you mind checking it out? :stuck_out_tongue_closed_eyes: Thank you!

**Comment by dex4er** [comment]
> thanks for commenting. I am checking the [official istio installation page for GKE](https://istio.io/latest/docs/setup/platform-setup/gke/) and they don't mention the same ports as you do. Are those ports new for istio 1.9? I can't see them anywhere.
> 
> Anyway, I am going to add the mention to the istio installation for GKE page to be more helpful, as you suggested.

Istio has incomplete documentation too and then there are some issues in this project:

https://github.com/istio/istio/issues/30661

https://github.com/istio/istio/issues/30646

As you can see, one issue was filled by me personaly.

**Review by lucasponce — APPROVED** [review]

**Pull request merged**